### DISCUSSION TITLE
DPL: Ist ein verzögertes Herunterregeln des Wechselrichters möglich ohne das Hochregeln zu beeinflussen?

### DISCUSSION THREAD

**Discussion by TriBoone** [hoylabs/OpenDTU-OnBattery#1079] [Q&A]
Hallo,

gibt es die Möglichkeit, eine Verzögerung einzustellen, damit die DTU den Wechselrichter erst nach x Sekunden herunterregelt?
(Das Hochregeln soll weiterhin schnell nach dem Abfrage-Intervall/dem Erhalt der Werte geschehen.)

Hintergrund: Da z. B. die Microwelle in Intervallen strahlt, wird der Wechselrichter getriggert hoch- und herunterzuregeln.
Hier suche ich einen Möglichkeit das **schnelle Hochregeln beizubehalten**, ABER das **Herunterregeln** zu **verzögern**, also erst herunterzuregeln, wenn x Sekunden vergangen sind und der Energieverbrauch gesunken ist.
Mein Ziel ist keine Nulleinspeisung. Ich möchte die maximale Einspeisung begrenzen, aber selbst das Maximum an Energieersparnis nutzen können.

**Comment by spcqike** [comment]
Nein sowas geht nicht in der Software.

Ich stelle es mir auch schwierig vor, zu erfassen, wann die Microwelle aus ist. "Verbrauch gesunken und x Sekunden vergangen" schön und gut. der Verbrauch kann ja aber auch durch einen ganz anderen weggefallenen Verbraucher gesunken sein.

Was mir auf die Schnelle einfällt als halbwegs probate Lösung: ein WLAN Zwischenstecker an der Microwelle, der deinen gewünschten Mehrverbrauch meldet, wenn die Microwelle läuft. das sollte mit jeder Tasmota-fähigen Lösung gehen.

Beispiel: Du willst 600W mehr einspeisen, solange die Mikrowelle läuft. Der Zwischenstecker meldet nun diese festen 600W, wenn die Mikrowelle aktiv ist (egal welche Leistung sie tatsächlich zieht). In den Powermeter Einstellungen nimmst du nun deinen echten Gesamtverbrauch auf Phase1 und den Zwischenstecker auf Phase2. In Summe hast du, solange die Mikrowelle läuft, nun den echten Verbrauch + 600W. Natürlich speißt du damit teilweise viel mehr ein, als notwendig. Immerhin kann es passieren, dass auch der echte Verbrauch die Mikrowelle enthält und du zusätzlich 600W erzeugst. Kommt halt auf die Abtastung an.

Eine zweite Möglichkeit wäre das anpassen des Targets um deinen Offset. bspw. Target 0W -> Mikrowele läuft -> Reduziere auf -600W. Auch damit speißt du immer die Last der Mikrowelle ein, und auch da kann und wird es passieren, dass du viel zu viel einspeist. Diese Umsetzung plane ich noch für die Waschmaschine, da der Motor doch viel Zappeln in der Last erzeugt, der Shelly einfach eine gewisse Trägheit mitbringt und es wenig Sinn macht, erst beim Erfassen einer kurzen Spitze (die aus dem Netz bedient wird) hochzuregeln und einzuspeisen, bis sie wieder weg ist und der Überschuss dann ins Netz geht. Lieber würd ich da direkt 100W mehr einspeise, solange der Überschuss aus der PV da ist, und den initialen Netzbezug sparen. Wie viel Wh das am Ende des Tages ausmacht .... mus sich zeigen. bei 1-2s Regelzeit sicherlich nicht viel.... aber naja :D mal gucken

Aber, warum willst du es so überhaupt machen? Was siehst du als Nachteil des schnellen Herunterregelns? Ist es nicht sogar gut, wenn schnell auf Lastwechsel reagiert wird? eine Mikrowelle hat doch, anders als der Motor der Waschmaschine, eine gute konstante Last für einige Sekunden, oder?

mit dem PR #1077 sind Abtastintervalle von 1s möglich. der Wechselrichter ist super schnell hoch- und auch runter gefahren. Solange die Taktzeit der Mikrowelle >~5s beträgt, sehe ich keinen Grund, warum man da nicht nachregeln sollte.

**Reply by TriBoone** [reply]
Danke für die rasche Antwort. 😃

Ich werde es mit einem schnelleren Abfrage-Intervall in der DTU versuchen und Rückmeldung geben.
Danke für den PR #1077 inkl. des Ablaufs für einen "4 MB Flashspeicher", das probiere ich aus. (Kann aber ein wenig dauern, bis ich dazu komme.)

Meine Wahrnehmung: Natürlich ist eine schnelle Reglung sinnvoll, aber durch das schnelle Herunterregeln wird mehr Energie aus dem Netz benötigt, da der Wechselrichter (HMS-1600-4T) erst hochregeln muss. Würde der für Zeit x hochgeregelt bleiben, würde weniger aus dem Netz benötigt. Das Mehr-Einspeisen während der hochgeregelten Zeit ist für mich ok.
Das passt auch zu deinem Beispiel mit der Waschmaschine. 😃
Die Intervall-Dauer unserer Mikrowelle scheint sich mit der eingestellten Stufe zu verändern. Die Last scheint beim Strahlen eher gleich zu bleiben, nur die Dauer von Strahlen und Nicht-Strahlen variiert. (Das kann ich aber noch genauer erkunden.)

Maximal 1,6 kW kann unser Balkonkraftwerk (kein Speicher vorhanden), die Microwelle zieht >1 kW, daher regelt der Wechselrichter meist voll auf, inklusive der vorhandenen Grundlast bei uns zu Hause.

Aktuelles Ziel liegt bei -600 W (also max. Einspeisen von 600 W) mit einer Hysterese von 50 W, dass funktioniert so gut für mich.

**Comment by Manos1966** [comment]
es waere Interessant zu wissen, wieviel Strom die Microwelle waehrend des gesamten Koch-Vorgangs verbraucht (in kWh).
Als ich das bei meinem gemessen habe, kam ich bei etwa 1,3 Cent pro Nutzung (Microwelle mit 1000W Leistung, 2 Minuten  Nutzung).
...ist nicht mal die Diskussion wert...

**Comment by schlimmchen** [comment]
> ...ist nicht mal die Diskussion wert...

Das sehe ich anders. Das Problem betrifft ja auch die gefürchteten Herdplatten. Und im Sommer, wenn der Akku eh zu klein ist weil so viel Energie vom Himmel kommt, ist es total ungeschickt, Energie zu verschwenden.

> Hier suche ich einen Möglichkeit das schnelle Hochregeln beizubehalten, ABER das Herunterregeln zu verzögern

Genau das wünsche ich mir auch.

> Ich stelle es mir auch schwierig vor

So kompliziert schätze ich das nicht ein. Ich hab folgende Idee im Kopf, die ich jetzt mal aufschreibe und zur Diskussion stelle:

Der DPL errechnet die von der PV-Anlage gewünschte Leistung, um den Haushaltsbedarf zu decken. Ist dieser Wert größer als der zuletzt berechnete Wert, dann gilt dieser sofort (und der WR erhält als Konsequenz ein neues Limit). Ist er aber kleiner, gilt er erstmal nicht. Um zu entscheiden, wann eine neue Leistung gelten soll, hält der DPL Buch darüber, wann er welche Leistung errechnet hat. Die in den letzten x (z.B. 10) Sekunden höchste errechnete Leistung gilt dann. Das hat natürlich zur Folge, dass erst nach x Sekunden *nachdem* die Mikrowelle oder der Herd tatsächlich nicht mehr genutzt wird, der WR zurückgefahren wird. Es hat aber den großen Vorteil, dass man dem Bedarf tatsächlich entgegensteuert, auch wenn man dafür in den Pausen und nachdem die Nutzung des Herdes vorrüber ist, etwas einspeist. Das spielt aber für viele im Sommer keine Rolle, weil "(Fast) Egal wie groß der Akku ist: Im Sommer ist er zu klein und um Winter ist er zu groß".

Und wenn wir vom Winter sprechen: Hier will ich jede unnütze Einspeisung verhindern. Also drehen wir die Logik um: Es gilt die *niedrigste* errechnete Leistung der letzten x Sekunden. D.h. es muss schon ein Staubsauger, oder die Wärmepumpe, oder was auch immer sein, das tatsächlich den Verbauch konstant mind. x Sekunden erhöht, bevor der WR mehr Leistung produziert. Sinkt der Verbrauch, wird dann auch sofort zurückgefahren.

Mir gefiele so ein Feature sehr gut. Hab ich einen Denkfehler oder könnte das gut funktionieren? (Das Feature ist natürlich nichts für Puristen, die glauben Einspeisen sei böse. Die lassen dann die Finger von diesem Feature.)

**Reply by TriBoone** [reply]
Für mich (ohne Speicher) wäre es im Winter das gleiche wie im Sommer - maximale Nutzung der Energie von den PV-Modulen im eigenen Haushalt, mit einem Herunterregeln nach x Sekunden nach Lastende (x Sekunden konfigurierbar). Die Einspeisung von Überschüssen ins Netz wäre mir egal.

@schlimmchen: Gute Beschreibung des gewünschten Features. 😃

**Reply by schlimmchen** [reply]
Magst du ein Feature-Request draus basteln? Da würde ich das Thema auch leichter wiederfinden...

**Reply by spcqike** [reply]
klingt sinnig. 
Woran macht man Sommer/Winter fest? Der Punkt, ab dem der Akku zu klein wird und nicht mehr voll geht, ist ja von Haushalt zu Haushalt und Akku zu Akku unterschiedlich. Oder hängt man die Logik an die Solar-Passthrough Kondition? Wenn Akku voll und PV Überschuss -> Nachlaufend, wenn nicht, vorlaufend. Und wenn, wie bekommt man es an Solar-only Systeme? :) da könnte es ja tendenziell immer "nachlaufend" sein, da man nichts speichern und aufheben kann.

Kleiner Denkanstoß dazu: 
deine Beschreibung würde, im Falle meines Gefrierschranks, dazu führen, dass für X Sekunden > 400W mehr eingespeist würden, als nötig. Einfach, weil der Anlaufstrom so extrem hoch ist. Je nachdem, wo der Stromzähler grad abgetastet wird, ist die Spitze mal größer, mal kleiner. 

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/dc33542d-2c25-4a4e-9608-0c5279b4f421)

Energetisch, grade im Sommer bei Überschuss, egal 😄 oder bei Solar-only. 

das "vorlaufen", also x Sekunden höhere Leistung bevor hochgeregelt wird, wäre im Falle meines Gefrierschranks bei Akkuanlagen sogar sinnvoll. die kurze Spitze kann man mit dem WR gar nicht abfangen, aufgrund der Mess- und Regelzeiten. Der Verbrauch ist wahrscheinlich schon lange wieder weg, bevor der WR die Leistung liefert.

**Reply by TriBoone** [reply]
> Magst du ein Feature-Request draus basteln? Da würde ich das Thema auch leichter wiederfinden...

@schlimmchen: Lege ich zeitnah an. 😃

**Reply by spcqike** [reply]
@TriBoone 
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/c0f70b6d-c23c-4ab0-b173-6ee2341067c3)

😄

**Reply by TriBoone** [reply]
@schlimmchen, @spcqike: Feature Request #1083
